### ISSUE TITLE
ACP events should have more info

### ISSUE THREAD

**Issue by dangerusslee** [NodeBB/NodeBB#6255]
- **NodeBB version:** v1.7.3
- **NodeBB git hash:** 0020b8a18b7ad9345162a5843817458724dafc14

- **Database type:** mongo
- **Database version:** v3.4.9

- **Exact steps to cause this issue:** I went to ACP to look at the events.  

- **What you expected:** I expected to be able to see more information about events, for example, when a post is edited, it should say that it was edited and what was changed (or at least the old and new fields.  

- **What happened instead:**

 Instead all I see is this:

```
{
    "oldTitle": "Free Hard Water Test Strip - Whirlpool and Morton",
    "newTitle": "Free Hard Water Test Strip - Morton",
    "timestamp": 1516228544053
}
```
There is no information as to what was edited (other than the title), and there's not even a link to the post.  

It does also provide the username, ip and date, but this is way too little information.  It would be really nice if the post could include a diff of what changed, like in Git, but at least the old and new fields would be good (for the post) and a link to the post in question (if possible)

**Comment by julianlam** [comment]
@dangerusslee We don't store diffs of posts right now, as that is a storage concern... every time you edit a post we'd have to store a second copy of it!

We intended to implement some post diffing instead, although this is a low priority at the moment. If there are other issues with ACP events, you'll have to specify which ones, and we can look into adding more information.

e.g. The post edit already shows the changed title, if applicable.

**Comment by dangerusslee** [comment]
@julianlam yes I understand that it's a storage concern, which is why I suggested diffing (but I understand this may take a while to implement).  

The point is there is no way to tell what happened to the post other than a title change.  I suggest making this an option (disabled by default) to allow storing of the post content as well.  

Some administrators may choose to use more storage vs not knowing what really happened to the post.

**Comment by barisusakli** [comment]
I think there is some confusion here, we only log `topic-rename` events right now and that shows the `oldTitle` `newTitle` we don't log post edit events at all. I agree events should have a link to the topic/post.

**Comment by dangerusslee** [comment]
@barisusakli Yes currently only topic-rename is logged, even if the title was never changed.  So if someone edits a post, and doesn't even touch the title, it just generates an event with both titles being completely the same.  

I've taken that as an assumption that the post was actually edited, but it would be nice if the post-rename didn't log things where the title didn't change, IN ADDITION to having something else logging the actual post edit (which may need to be enabled manually since it would be a storage hog).  Long term, of course, I would expect some sort of diff to be stored and it being enabled by default for everyone (since space won't be much of a concern anymore).

**Comment by barisusakli** [comment]
Sounds like a bug, if title didn't change there should be no topic-rename event, I will take a look

**Comment by barisusakli** [comment]
>@barisusakli Yes currently only topic-rename is logged, even if the title was never changed. So if someone edits a post, and doesn't even touch the title, it just generates an event with both titles being completely the same.

I can't reproduce this, can you give detailed repro steps and nodebb git hash?

**Comment by dangerusslee** [comment]
@barisusakli The has is included above.  

It seems to only happen with special characters.  Example:

```
{
    "oldTitle": "test &amp;amp; testing",
    "newTitle": "test &amp;amp; testing",
    "timestamp": 1516378026159
}
```

**Comment by barisusakli** [comment]
Ok fixed the issue with titles that have special characters.

**Comment by dangerusslee** [comment]
@barisusakli Thank you, except now I will know even less about what the users are doing.  Can we at least track edits that are done by a third party (like if the owner or a moderator edits someone's post?)

**Comment by barisusakli** [comment]
By owner I think you mean the administrator and not the post owner. We can log the post edit if it is edited by someone else other than the post owner.

**Comment by nodebb-misty** [comment]
Issue closed due to inactivity.

----

*This is an automated message. If you feel this action was in error, please comment on this issue so it can be looked at again*

**Comment by antoine-pous** [comment]
Bump!
